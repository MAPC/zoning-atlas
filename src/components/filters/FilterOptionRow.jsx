import React from 'react'

const FilterOptionRow = ({filterValue, index, option, setFilter}) => {
  return (
    <li
      data-value={option}
      className={filterValue && filterValue.includes(option) ? 'filters__option filters__option--active' : 'filters__option'}
      key={`option-${index}`}
      onClick={(e) => {
        const selectedOption = e.currentTarget.querySelector('.filters__value').innerText;
        if (filterValue === undefined) {
          setFilter([selectedOption])
        } else if (!filterValue.includes(selectedOption)) {
          setFilter(filterValue.concat([selectedOption]))
        } else if (filterValue.includes(selectedOption) && filterValue.length > 1) {
          const selectedIndex = filterValue.indexOf(selectedOption)
          setFilter(filterValue.slice(0, selectedIndex).concat(filterValue.slice(selectedIndex+1)))
        } else {
          setFilter(undefined)
        }
      }}>
        <span className={filterValue && filterValue.includes(option) ? 'filters__value filters__value--active' : 'filters__value'}>{option}</span>
        <button className={filterValue && filterValue.includes(option) ? 'filters__button filters__button--active' : 'filters__button'}>+</button>
      </li>
  )
}

export default FilterOptionRow
