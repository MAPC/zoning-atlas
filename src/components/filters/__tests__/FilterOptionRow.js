import React from 'react';
import renderer from 'react-test-renderer';
import { describe, it, expect } from '@jest/globals';

import FilterOptionRow from '../FilterOptionRow';

describe('Filter option (row)', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<FilterOptionRow index={0} option="Acton" setFilter={() => null}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
