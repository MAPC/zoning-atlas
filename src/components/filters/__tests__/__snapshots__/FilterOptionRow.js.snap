// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter option (row)  renders correctly 1`] = `
<li>
  <div
    className="filters__option"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex="0"
  >
    <span
      className="filters__value"
    />
    <span
      className="filters__button"
    >
      +
    </span>
  </div>
</li>
`;

exports[`Filter option (row) renders correctly 1`] = `
<li>
  <div
    className="filters__option"
    data-value="Acton"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex="0"
  >
    <span
      className="filters__value"
    >
      Acton
    </span>
    <span
      className="filters__button"
    >
      +
    </span>
  </div>
</li>
`;
